File: 백준/Silver/10773. 제로/제로.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int K = sc.nextInt();

        Stack<Integer> stack = new Stack<>();

        for(int i = 0; i < K; i++){
            int num = sc.nextInt();
            if(num == 0){
                stack.pop();
            }
            else {
                stack.push(num);
            }
        }
        sc.close();

        int sum = 0;
        while(!stack.empty()){
            sum += stack.pop();
        }
        System.out.println(sum);
    }
}

File: 백준/Silver/10773. 제로/제로.kt
import java.util.*
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val K = readLine().toInt()
    val stack = Stack<Int>()

    repeat(K) {
        val num = readLine().toInt()

        if (num == 0) stack.pop() else stack.push(num)
    }
    close()

    val bw = BufferedWriter(OutputStreamWriter(System.out))
    bw.write(stack.sum().toString())
    bw.flush()
    bw.close()
}